pkg/util/tls/tls_integration_test.go

```diff
@@ -0,0 +1,484 @@
+package tls_test
```

Comment:
I would keep it simple. Our `integration` tests are mainly focused on whenever you need to run the `cortex` binary, which is not the case of these tests. These tests are closer to unit tests then integration tests (even if you start an HTTP/gRPC server, but it's something we do in many unit tests) so I would keep them here.

```suggestion
package tls
```

Reply:
That makes all perfect sense. I think I can't easily change the package to `tls` as this would lead to an import path cycle through the used `grpcclient`.

Reply:
What's about a `test` subpackage?

Reply:
Very good point, move it there